rfcs/0002-reimplement-start-command.md

```diff
@@ -0,0 +1,109 @@
+# Reimplement Yarn Start Command
+
+## Proposal
+
+The buildpack will no longer call `yarn start` directly, but instead will
+reimplement the behavior of `yarn start` directly.
+
+## Motivation
+
+We've seen flakey issues with process signal handling once we implemented [RFC
+0001](0001-start-command.md). Specifically, on some container managers, it
+appears that `tini` is not waiting for the entire process group to exit before
+exiting itself. This means that our tests that make assertions that log lines
+appear in the output during process shutdown are unreliable.
+
+Furthermore, the maintainers of the `yarn` CLI have [openly
+stated](https://github.com/yarnpkg/yarn/issues/4667#issuecomment-628241114)
+that this issue is considered resolved if you upgrade to `yarn` v2, which the
+buildpack has not. There is currently no backport of this functionality to v1
+and may never be.
+
+Given all of this, it seems to make more sense now to just take on the cost of
+reimplementing the behavior of `yarn start` and removing the complexity of
+using `yarn` and `tini` together.
+
+## Implementation
+
+The buildpack will still require `node` and `node_modules` during the `launch`
+phase, but will no longer need to require `yarn` or `tini`.
+
+Detection will continue to pass once a `package.json` file is present in the
```

Comment:
[RFC 0001](https://github.com/paketo-buildpacks/yarn-start/blob/66eb4104532e4667e7813264b8d85bad205c9f99/rfcs/0001-start-command.md#implementation-optional) (which the current implementation is based on) says `Detection will pass once a yarn.lock file is present`. Any reason to change that criterion?

Based on [yarn documentation](https://classic.yarnpkg.com/en/docs/managing-dependencies#toc-adding-a-dependency), it looks like yarn users will already have a generated `yarn.lock` file as part of their app. Maybe detect on `yarn.lock`; and if `package.json` is absent, default to `node server.js`?

Reply:
No. This is a mistake. I cribbed most of this from the `npm-start` buildpack RFC and I guess I missed this part.